x/evm/keeper/utils.go

```diff
@@ -0,0 +1,15 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/evmos/evmos/v18/x/evm/types"
+)
+
+// IsContract determines if the given address is a smart contract.
+func (k *Keeper) IsContract(ctx sdk.Context, addr common.Address) bool {
+	return !types.BytesAreEmptyCodeHash(k.GetCodeHash(ctx, addr).Bytes())
```

Comment:
as this is a keeper method it's actually unnecessarily complex and we should be able to just check if the store has an entry